### PR TITLE
Fix segfault for empty @return tags in modifiers

### LINKED ISSUE #10433: [DocStringAnalyser] Null ptr dereference in copyMissingTags()
## Description

```
contract C {
        ///@return
        modifier m22 {}
}

contract D is C {
        modifier m22 {}
}
```

references a nullptr here

https://github.com/ethereum/solidity/blob/887569efd5671d9235c29c57229e64a700682e28/libsolidity/analysis/DocStringAnalyser.cpp#L69

because `returnParamters()` is null.

## Environment

- Compiler version: latest develop

## Steps to Reproduce

```
$ solc test.sol
```

### PR DESCRIPTION
**Pull request by Marenz**
fixes #10433

### CODE REVIEW

**Comment by chriseth** [comment]
Is this only for modifiers? Can we fix it properly and disallow `@returns` for modifiers?

File: test/libsolidity/syntaxTests/natspec/invalid/docstring_inherit_modifier_no_return.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	///@return
+	modifier m22 {}
+}
+
+contract D is C {
+	modifier m22 {}
+}
+// ----
+// SyntaxError 2883: (39-41): Modifier body does not contain '_'.
+// SyntaxError 2883: (77-79): Modifier body does not contain '_'.
```
Comment (leonardoalt) [inline test/libsolidity/syntaxTests/natspec/invalid/docstring_inherit_modifier_no_return.sol]: Maybe add `_` in the modifiers to get rid of those errors, since they're unrelated?

File: test/libsolidity/syntaxTests/natspec/invalid/docstring_inherit_modifier_no_return.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// DocstringParsingError 6546: (14-24): Documentation tag @return not valid for modifiers.
+// TypeError 9456: (64-79): Overriding modifier is missing "override" specifier.
+// TypeError 4334: (26-41): Trying to override non-virtual modifier. Did you forget to add "virtual"?
```
Comment (leonardoalt) [inline test/libsolidity/syntaxTests/natspec/invalid/docstring_inherit_modifier_no_return.sol]: Also these 2 errors seem unrelated?

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 3]
```diff
@@ -66,7 +66,11 @@ void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, Stru
 				size_t docParaNameEndPos = content.content.find_first_of(" \t");
 				string const docParameterName = content.content.substr(0, docParaNameEndPos);
 
-				if (docParameterName != _declaration->returnParameters().at(n)->name())
+				if (
+					_declaration->returnParameterList() &&
```
Comment (hrkrshnn) [inline libsolidity/analysis/DocStringAnalyser.cpp]: Shouldn't `_declaration->returnParameterList()` be an assertion?

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 4]
```diff
@@ -66,7 +66,11 @@ void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, Stru
...
+					_declaration->returnParameters().size() > n &&
```
Comment (hrkrshnn) [inline libsolidity/analysis/DocStringAnalyser.cpp]: Won't this https://github.com/ethereum/solidity/blob/83fc7d3f7de5d8c0ce184dab4a79e2248b930c89/libsolidity/analysis/DocStringTagParser.cpp#L181 be triggered before? Can we move this into an assertion?

Comment (Marenz) [reply libsolidity/analysis/DocStringAnalyser.cpp]: No, this won't trigger because what we have is not a `FunctionDefinition` but a `ModifierDefinition`

File: libsolidity/analysis/DocStringAnalyser.cpp (hunk 3 above)
Comment (Marenz) [reply libsolidity/analysis/DocStringAnalyser.cpp]: I don't think so. The latest version triggers an error already in the TagParser, but not a fatal one so this could still receive unvalidated data..

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 5]
```diff
@@ -42,7 +42,14 @@ void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, Stru
 	if (_baseFunctions.size() != 1)
 		return;
 
-	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>((*_baseFunctions.begin())->annotation());
+	auto hasParameter = [](CallableDeclaration const& __declaration, size_t _n)
+	{
+		return __declaration.returnParameterList() &&
```
Comment (chriseth) [inline libsolidity/analysis/DocStringAnalyser.cpp]: double underscore prefix is reserved. Weird that the compiler does not warn...

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 6]
```diff
@@ -42,7 +42,14 @@ void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, Stru
...
+			__declaration.returnParameters().size() > _n;
+	};
+
+	auto& baseFunction = **_baseFunctions.begin();
```
Comment (chriseth) [inline libsolidity/analysis/DocStringAnalyser.cpp]: Can you move this directly after the `if (_baseFunction.size()` - it makes total sense there. And please also add an explicit type, since this is one of the most important local variables.

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 7]
```diff
@@ -42,7 +42,14 @@ void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, Stru
 	if (_baseFunctions.size() != 1)
 		return;
 
-	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>((*_baseFunctions.begin())->annotation());
+	auto hasParameter = [](CallableDeclaration const& __declaration, size_t _n)
```
Comment (chriseth) [inline libsolidity/analysis/DocStringAnalyser.cpp]:
```suggestion
	auto hasReturnParameter = [](CallableDeclaration const& __declaration, size_t _n)
```

**Review by chriseth — DISMISSED** [review]
